mcodingbot/tasks/stats.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from dataclasses import dataclass
+from math import log
+from typing import TYPE_CHECKING, Any
+
+from mcodingbot.config import CONFIG
+
+if TYPE_CHECKING:
+    from mcodingbot.bot import Bot
+
+
+LOGGER = logging.getLogger(__file__)
+
+
+async def loop_update_channels(bot: Bot) -> None:
+    while True:
+        try:
+            await update_channels(bot)
+        except Exception:
+            LOGGER.error("Failed to update channel stats:", exc_info=True)
+
+        await asyncio.sleep(5 * 60)
+
+
+async def update_channels(bot: Bot) -> None:
+    stats = await get_stats(bot)
+
+    sub_channel = bot.cache.get_guild_channel(CONFIG.sub_count_channel)
+    view_channel = bot.cache.get_guild_channel(CONFIG.view_count_channel)
+    member_channel = bot.cache.get_guild_channel(CONFIG.member_count_channel)
+
+    assert sub_channel
+    assert view_channel
+    assert member_channel
```

Comment:
i'm not a fan of having `assert` elsewhere than within unittests tbh
also since theses lines are basically the same repeated 3 times, why not extracting it to a method?

Reply:
Removed the asserts, they were no longer needed.